engine/src/test/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStoreSpec.scala

```diff
@@ -244,22 +254,130 @@ class SqlWorkflowStoreSpec extends AnyFlatSpec with CromwellTimeoutSpec with Mat
+        _ = startableWorkflows3.size should be(0)
+
+        // hog group "Zardoz" has tokens to run workflows, hence don't exclude it
+        startableWorkflows4 <- workflowStore.fetchStartableWorkflows(5, "A08", 1.minute, Set.empty[String])
+        _ = startableWorkflows4.map(_.hogGroup.value).toSet.head should be("Zardoz")
+        _ = updateWfToRunning(startableWorkflows4)
+
+        startableWorkflows5 <- workflowStore.fetchStartableWorkflows(5, "A08", 1.minute, Set.empty[String])
+        _ = startableWorkflows5.map(_.hogGroup.value).toSet.head should be("Zardoz")
+        _ = updateWfToRunning(startableWorkflows5)
+
+        // remove entries from WorkflowStore
+        workflowsList = startableWorkflows1 ++ startableWorkflows2 ++ startableWorkflows3 ++ startableWorkflows4 ++ startableWorkflows5
+        _ = workflowsList.foreach(wf => Await.result(workflowStore.deleteFromStore(wf.id), 5.seconds))
+      } yield()).futureValue
+    }
+
+    it should "select workflows from hog groups to start in round robin approach for multiple hog groups" taggedAs DbmsTest in {
+      // first submission of 10 workflows for hogGroup "Goldfinger"
+      for (_ <- 1 to 10) {
+        workflowStore.add(includedGroupSourceFilesCollection1)
+      }
+      Thread.sleep(5.seconds.toMillis)
+      // second submission of 10 workflows for hogGroup "Highlander"
+      for (_ <- 1 to 15) {
+        workflowStore.add(includedGroupSourceFilesCollection2)
+      }
+
       (for {
-        submissionResponsesExcluded <- workflowStore.add(excludedGroupSourceFilesCollection)
-        submissionResponsesIncluded1 <- workflowStore.add(includedGroupSourceFilesCollection1)
-        submissionResponsesIncluded2 <- workflowStore.add(includedGroupSourceFilesCollection2)
-        submissionResponsesIncluded3 <- workflowStore.add(includedGroupSourceFilesCollection3)
-        startableWorkflows <- workflowStore.fetchStartableWorkflows(3, "A08", 1.second, excludedGroups = Set("Zardoz"))
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesExcluded.map(_.id).toList).size should be(0)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded1.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded2.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded3.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).size should be(3)
-        _ <- workflowStore.deleteFromStore(submissionResponsesExcluded.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded1.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded2.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded3.head.id) // Tidy up
```

Comment:
I had to remove this because now we fetch workflows from only one hog group at a time. But I have tried to cover this scenario in the added test cases.